middleware/with-admin.ts

```diff
@@ -0,0 +1,34 @@
+import { NextApiHandler } from "next";
+import getConfig from "next/config";
+
+import { NOT_ADMIN_WALLET } from "helpers/constants";
+import { toLower } from "helpers/string";
+
+import { siweMessageService } from "services/ethereum/siwe";
+import { Logger } from "services/logging";
+
+import { isMethodAllowed } from "server/utils/http";
+
+const { publicRuntimeConfig } = getConfig();
+
+Logger.changeActionName(`withAdmin()`);
+
+export const withAdmin = (handler: NextApiHandler, allowedMethods = ["GET"]): NextApiHandler => {

```

Comment:
supposedly we gave the `admin` role to this session, so we could simply check for the existance of that role on the session and return false if doesn't exist